library/core/src/main/java/com/google/android/exoplayer2/source/MediaSource.java

```diff
@@ -275,6 +275,11 @@ void prepareSource(
    */
   void releasePeriod(MediaPeriod mediaPeriod);
 
+  /**
+   * Returns the tag set on media source, or null when none was set.
+   */
+  Object getTag();
```

Comment:
Can you move the method somewhere else to keep the release methods bundled together? For example, above addEventListener or below removeEventListener.

Reply:
Also, can you make this a default method which returns null for now? That prevents breaking other existing implementations.

Reply:
@Nullable is missing here too.